controllers/heatapi_controller.go

```diff
@@ -361,6 +361,8 @@ func (r *HeatAPIReconciler) reconcileDelete(ctx context.Context, instance *heatv
 	controllerutil.RemoveFinalizer(instance, helper.GetFinalizer())
 	r.Log.Info("Reconciled API delete successfully")
 
+	instance.Status.ObservedGeneration = instance.Generation
```

Comment:
As per [1] we are trying to invert the pattern and bump `ObservedGeneration` at the beginning of the Reconciliation loop. This should be basically moved at the beginning of `reconcileNormal` [2] (more details in [1] and [3]).
The top level, as done in [1], should check the subResources ObservedGeneration before marking ReadyCondition as True.
Let' sync up and update this patch as long as [1] is finalized, so we can spread the pattern across the board.
(@dprince FYI ^) 

[1] https://github.com/openstack-k8s-operators/glance-operator/pull/501
[2] https://github.com/openstack-k8s-operators/heat-operator/blob/main/controllers/heatapi_controller.go#L552
[3] https://github.com/openstack-k8s-operators/docs/pull/102

Reply:
Sorry mate, apologies. I missed all of these comments, need a better email filtering system I think.

Ok, I have moved all of these now, did you want to take another look and make sure I have captured your intentions?

Reply:
Actually, now they're in the right place. So after the init of the other conditions.